src/main/java/jarvis/Parser.java

```diff
@@ -0,0 +1,134 @@
+     */
+    public static Command parseCommand(final String userInput) throws InvalidCommandException {
+        String[] userInputSpilt = userInput.split(" ");
+
+        if (userInput.equalsIgnoreCase("bye")) {
+            return new ByeCommand();
+        } else if (userInput.equalsIgnoreCase("list")) {
+            return new ListCommand();
+        } else if (userInputSpilt[0].startsWith("mark")) {
+            return new MarkCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("unmark")) {
+            return new UnmarkCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("delete")) {
+            return new DeleteCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("find")) {
+            return new FindCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("todo")) {
+            return new TodoCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("deadline")) {
+            return new DeadlineCommand(userInput);
+        } else if (userInputSpilt[0].equalsIgnoreCase("event")) {
+            return new EventCommand(userInput);
+        } else {
+            throw new InvalidCommandException(null);
+        }
+    }
+
+    /**
+     * Parses a date and time string into a LocalDateTime object.
+     *
+     * @param inputDateTime The date and time string to parse.
+     * @return A LocalDateTime object representing the parsed date and time.
+     * @throws InvalidDateTimeFormatException If the input date and time format is
+     *                                        invalid.
+     */
+    public static LocalDateTime parseDateTime(final String inputDateTime) throws InvalidDateTimeFormatException {
+        List<String> inputFormats = new ArrayList<>();
+        inputFormats.add(Ui.DATE_TIME_FORMAT);
+        inputFormats.add("dd MMM yyyy HHmm");
+        inputFormats.add("yyyy-MM-dd HHmm");
+        inputFormats.add("dd/MM/yyyy HHmm");
+
+        LocalDateTime result = null;
+        for (String inputFormat : inputFormats) {
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern(inputFormat);
+                LocalDateTime parsedDateTime = LocalDateTime.parse(inputDateTime, formatter);
+                result = parsedDateTime;
+                break;
+            } catch (Exception e) {
+                continue;
+            }
+        }
+
+        if (result != null) {
+            return result;
+        } else {
+            throw new InvalidDateTimeFormatException(inputDateTime);
+        }
+    }
+
+    /**
+     * Parses a string representation of a task into a Task object.
+     *
+     * @param line The string representation of the task.
+     * @return A Task object representing the parsed task.
+     * @throws InvalidTaskFormatException If the task format is invalid.
+     */
+    public static Task parseStringToTask(final String line) throws InvalidTaskFormatException {
```

Comment:
If I read correctly, this should be the method that reads from the data file and converts the string to task? If so, would it be better to name it `decodeStringToTask` to avoid confusion with the method that parses the user input?
```suggestion
    public static Task encodeStringToTask(final String line) throws InvalidTaskFormatException {
```